be/src/exec/es/es_scroll_parser.cpp

```diff
@@ -519,7 +521,178 @@ Status ScrollParser::fill_columns(const TupleDescriptor* tuple_desc,
+                    array.push_back(val);
+                    break;
+                }
+                case TYPE_BIGINT: {
+                    int64_t val;
+                    RETURN_IF_ERROR(
+                            get_int_value<int64_t>(sub_col, sub_type, &val, pure_doc_value));
+                    array.push_back(val);
+                    break;
+                }
+                case TYPE_LARGEINT: {
+                    __int128 val;
+                    RETURN_IF_ERROR(
+                            get_int_value<__int128>(sub_col, sub_type, &val, pure_doc_value));
+                    array.push_back(val);
+                    break;
+                }
+                case TYPE_FLOAT: {
+                    float val;
+                    RETURN_IF_ERROR(
+                            get_float_value<float>(sub_col, sub_type, &val, pure_doc_value));
+                    array.push_back(val);
+                    break;
+                }
+                case TYPE_DOUBLE: {
+                    double val;
+                    RETURN_IF_ERROR(
+                            get_float_value<double>(sub_col, sub_type, &val, pure_doc_value));
+                    array.push_back(val);
+                    break;
+                }
+                case TYPE_BOOLEAN: {
+                    if (sub_col.IsBool()) {
+                        array.push_back(sub_col.GetBool());
+                        break;
+                    }
+
+                    if (sub_col.IsNumber()) {
+                        array.push_back(sub_col.GetInt());
+                        break;
+                    }
+
+                    bool is_nested_str = false;
+                    if (pure_doc_value && sub_col.IsArray() && sub_col[0].IsBool()) {
+                        array.push_back(sub_col[0].GetBool());
+                        break;
+                    } else if (pure_doc_value && sub_col.IsArray() && sub_col[0].IsString()) {
+                        is_nested_str = true;
+                    } else if (pure_doc_value && sub_col.IsArray()) {
+                        return Status::InternalError(ERROR_INVALID_COL_DATA, "BOOLEAN");
+                    }
+
+                    const rapidjson::Value& str_col = is_nested_str ? sub_col[0] : sub_col;
+
+                    const std::string& val = str_col.GetString();
+                    size_t val_size = str_col.GetStringLength();
+                    StringParser::ParseResult result;
+                    bool b = StringParser::string_to_bool(val.c_str(), val_size, &result);
+                    RETURN_ERROR_IF_PARSING_FAILED(result, str_col, type);
+                    array.push_back(b);
+                    break;
+                }
+                // date/datetime v2 is the default type for catalog table,
+                // see https://github.com/apache/doris/pull/16304
+                // No need to support date and datetime types.
+                case TYPE_DATEV2: {
```

Comment:
Old external table(not catalog external table) will still use the old date type

Reply:
As we discussed, we do not maintain the external table any more.